# the planner will integrate local maps into a global map
global_map:
  resolution: 0.40 # [m] voxel edge size

planning_tree:
  resolution: 0.4

distance_transform:
  submap_distance: 20.0 # [m] from the UAV into all directions

main_timer:
  rate: 20.0 # [Hz]

diagnostics_timer:
  rate: 10.0 # [Hz]

future_check_timer:
  rate: 10.0 # [Hz]

euclidean_distance_cutoff: 1.0 # [m]
safe_obstacle_distance: 1.0 # [m]

distance_penalty: 1.0
greedy_penalty: 1.0

max_waypoint_distance: 0.0 # [m] // or set negative for no padding
min_altitude: -1000.0 # [m]
max_altitude: 1000.0 # [m]
unknown_is_occupied: true
replan_after: 1.0 # [s]

timeout_threshold: 5.0 # [s] maximum planning time
time_for_trajectory_generator: 0.5 # [s]
trajectory_generator_input_length: 15.0 # [s] (the rest of path will be ignored during trajectory generation)

use_map_from_file: true

subt_planner: 
  use: true
  apply_postprocessing: false
  make_path_straight: true # cannot be applied if postprocessing is active
  debug_info: false
  clearing_dist: 1.0
  max_obst_dist: 5.0 # max dist from nearest obstacle (required for reliable localization)
  admissibility: 2.0 # "suboptimality factor" - 2.0 allows upto 2 times longer paths than optimal path  

  apply_pruning: true
  pruning_dist: 0.4
  
  planning_tree: 
    bounding_box: 
      horizontal: 60.0 # (m)
      vertical: 30.0 # (m)

  postprocessing: 
    safe_dist: 2.5 # (m)
    max_iteration: 5
    horizontal_neighbors_only: true
    z_diff_tolerance: 0.5 # (m)
    fix_goal_point: true

  shortening: 
    window_size: 20
    distance: 1.0 # (m)

# visualization
points_scale: 0.05
lines_scale: 0.02
